pkg/controller/context.go

```diff
@@ -322,21 +323,28 @@ func NewContextFactory(ctx context.Context, opts ContextOptions) (*ContextFactor
 func (c *ContextFactory) Build(component ...string) (*Context, error) {
 	restConfig := util.RestConfigWithUserAgent(c.baseRestConfig, component...)
 
+	scheme := runtime.NewScheme()
+	for _, err := range []error{
+		kscheme.AddToScheme(scheme),
+		cmscheme.AddToScheme(scheme),
+		gwscheme.AddToScheme(scheme),
+	} {
+		if err != nil {
+			return nil, fmt.Errorf("error adding scheme to runtime: %w", err)
+		}
+	}
+
 	clients, err := buildClients(restConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	// Create event broadcaster.
-	// Add cert-manager types to the default Kubernetes Scheme so Events can be
-	// logged properly.
```

Comment:
This comment helps future maintainers know why we added these schemes here.
And if we ever need to attach events to other API Kinds it will be reminder that their schemes need adding here too.
Please re-instate it and move it near the schema setup code,
or explain why you think the comment is not useful.